ui/app/selectors/permissions.js

```diff
@@ -261,6 +263,21 @@ export function getLastConnectedInfo (state) {
   }, {})
 }
 
+export function getPermissionsMetadataHostCounts (state) {
+  const metadata = getPermissionDomainsMetadata(state)
+  return Object.values(metadata).reduce((acc, val) => {
+    const { host } = val
+    if (host) {
+      if (!acc[host]) {
+        acc[host] = 1
+      } else {
+        acc[host] += 1
+      }
+    }
+    return acc
+  }, {})
```

Comment:
Nit: It would be better to avoid generic variable names where possible

```suggestion
  return Object.values(metadata).reduce((counts, { host }) => {
    if (host) {
      if (!counts[host]) {
        counts[host] = 1
      } else {
        counts[host] += 1
      }
    }
    return counts
  }, {})
```